File: backend/database/models/device.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, func
from backend.database.base import Base

class Device(Base):
    __tablename__ = "devices"

    id = Column(Integer, primary_key=True, autoincrement=True)
    mac_address = Column(String, unique=True, nullable=False, index=True)
    ip_address = Column(String, index=True)
    hostname = Column(String)
    vendor = Column(String)
    device_type = Column(String, default="unknown")
    custom_name = Column(String)
    icon_key = Column(String, default="device")
    notes = Column(String)
    is_online = Column(Boolean, default=False, index=True)
    is_blocked = Column(Boolean, default=False)
    is_protected = Column(Boolean, default=False)
    bandwidth_limit_kbps = Column(Integer, default=0)
    first_seen = Column(DateTime, server_default=func.now())
    last_seen = Column(DateTime, server_default=func.now(), onupdate=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    @property
    def display_name(self) -> str:
        return self.custom_name or self.hostname or f"Device ({self.mac_address[-8:]})"

File: frontend/src/types/device.ts
export interface Device {
  id: number
  mac_address: string
  ip_address: string | null
  hostname: string | null
  vendor: string | null
  device_type: string
  display_name: string
  custom_name: string | null
  icon_key: string
  notes: string | null
  is_online: boolean
  is_blocked: boolean
  is_protected: boolean
  bandwidth_limit_kbps: number
  first_seen: string | null
  last_seen: string | null
}

export interface DevicesResponse {
  devices: Device[]
  total: number
}
